app/uk/gov/hmrc/securemessage/controllers/SecureMessageController.scala

```diff
@@ -143,4 +136,21 @@ class SecureMessageController @Inject()(
           }
         }
   }
+
+  private def handleCreateConversationErrors(conversationId: String, error: Exception): Result = {
+    val errMsg = s"Error on conversation with id $conversationId: ${error.getMessage}"
+    logger.error(error.getMessage, error.getCause)
+    error match {
+      case ee: EmailError                 => Created(Json.toJson(ee.message))
+      case ee: NoReceiverEmailError       => Created(Json.toJson(ee.message))
+      case de: DuplicateConversationError => Conflict(Json.toJson(de.message))
+      case ih: InvalidHtmlContent         => BadRequest(Json.toJson(ih.message))
```

Comment:
love this pattern, it is so clear what the logic is and it is all in one place :) so just a praise here. 
@jim-collins - you see, the pattern is followed, you are the hero now :)